libs/services/designer-client-services/src/lib/base/operationManifest.ts

```diff
@@ -0,0 +1,641 @@
+const slidingwindow = 'slidingwindow';
+const wait = 'wait';
+const delay = 'delay';
+const delayuntil = 'delayuntil';
+const http = 'http';
+const httpwebhook = 'httpwebhook';
+const httpaction = 'httpaction';
+const httptrigger = 'httptrigger';
+const httpswaggeraction = 'httpswaggeraction';
+const httpswaggertrigger = 'httpswaggertrigger';
+const httpwebhookaction = 'httpwebhookaction';
+const httpwebhooktrigger = 'httpwebhooktrigger';
+const initializevariable = 'initializevariable';
+const setvariable = 'setvariable';
+const incrementvariable = 'incrementvariable';
+const decrementvariable = 'decrementvariable';
+const appendtoarrayvariable = 'appendtoarrayvariable';
+const appendtostringvariable = 'appendtostringvariable';
+
+export const azureFunctionConnectorId = '/connectionProviders/azureFunctionOperation';
+const dataOperationConnectorId = 'connectionProviders/dataOperationNew';
+const controlConnectorId = 'connectionProviders/control';
+const dateTimeConnectorId = 'connectionProviders/datetime';
+const scheduleConnectorId = 'connectionProviders/schedule';
+const httpConnectorId = 'connectionProviders/http';
+const variableConnectorId = 'connectionProviders/variable';
+
+const supportedManifestTypes = [
+  appendtoarrayvariable,
+  appendtostringvariable,
+  as2Encode,
+  as2Decode,
+  compose,
+  condition,
+  decrementvariable,
+  expression,
+  foreach,
+  function_,
+  http,
+  httpwebhook,
+  initializevariable,
+  incrementvariable,
+  invokefunction,
+  javascriptcode,
+  join,
+  liquid,
+  parsejson,
+  query,
+  recurrence,
+  request,
+  response,
+  select,
+  setvariable,
+  slidingwindow,
+  switchType,
+  serviceprovider,
+  table,
+  workflow,
+  xmlvalidation,
+  xslt,
+  flatfiledecoding,
+  flatfileencoding,
+  scope,
+  swiftdecode,
+  swiftencode,
+  terminate,
+  until,
+  wait,
+];
+
+export type getAccessTokenType = () => Promise<string>;
+
+export interface BaseOperationManifestServiceOptions {
+  apiVersion: string;
+  baseUrl: string;
+  httpClient: IHttpClient;
+  supportedTypes?: string[];
+}
+
+export class BaseOperationManifestService implements IOperationManifestService {
+  constructor(readonly options: BaseOperationManifestServiceOptions) {
+    const { apiVersion, baseUrl, httpClient } = options;
+    if (!apiVersion) {
+      throw new ArgumentException('apiVersion required');
+    } else if (!baseUrl) {
+      throw new ArgumentException('baseUrl required');
+    } else if (!httpClient) {
+      throw new ArgumentException('httpClient required');
+    }
+  }
+
+  isSupported(operationType: string, _operationKind?: string): boolean {
+    const { supportedTypes } = this.options;
+    const normalizedOperationType = operationType.toLowerCase();
+    return supportedTypes
+      ? supportedTypes.indexOf(normalizedOperationType) > -1
+      : supportedManifestTypes.indexOf(normalizedOperationType) > -1;
+  }
+
+  async getOperationInfo(definition: any, isTrigger: boolean): Promise<OperationInfo> {
```

Comment:
This operation should be abstract and should be implemented by standard and consumption separately, reason being serviceprovider is not a valid operation in consumption.